src/extensions/wcf/Microsoft.DotNet.UpgradeAssistant.Extensions.WCFUpdater/UpdaterFactory.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.WCFUpdater
+{
+    public class UpdaterFactory
+    {
+        public static PackageUpdater? GetPackageUpdater(string path, ILogger logger)
+        {
+            try
+            {
+                XDocument doc = XDocument.Load(path);
+                PackageUpdater packageUpdater = new PackageUpdater(doc, logger);
+                return packageUpdater;
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, $"Error loading project file {path}");
+                return null;
+            }
+        }
+
+        public static ConfigUpdater? GetConfigUpdater(string path, ILogger logger)
+        {
+            try
+            {
+                XDocument doc = XDocument.Load(path);
+                ConfigUpdater configUpdater = new ConfigUpdater(doc, logger);
+                return configUpdater;
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, $"Error loading project file {path}");
+                return null;
+            }
+        }
+
+        public static SourceCodeUpdater? GetSourceCodeUpdater(string path, Dictionary<string, object> context, ILogger logger)
+        {
+            try
+            {
+                SyntaxTree tree = CSharpSyntaxTree.ParseText(File.ReadAllText(path));
+                SourceCodeUpdater sourceCodeUpdater = new SourceCodeUpdater(tree, UpdateTemplateCode(context, logger), logger);
+                return sourceCodeUpdater;
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, $"Error parsing project file {path}");
+                return null;
+            }
+        }
+
+        public static List<SourceCodeUpdater>? GetDirectiveUpdaters(IEnumerable<string> path, ILogger logger)
```

Comment:
The name for this method is confusing. It's getting source code updaters but it's called GetDirectiveUpdaters.

Reply:
It's a different type of SourceCodeUpdater! It's for .cs files that only needs to add the using directives for coreWCF, such as BeanTrader.cs. I could change it to a different name to be more clear?

Reply:
The using directive updaters are constructed by a different constructor in the` SourceCodeUpdater` class and they can only be used to update directives, and can't do template code/remove old code stuff because in their constructor no template is passed in!